### ISSUE TITLE
rename twisted.python.logger to twisted.logger

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7652]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#7652|
|Type|task|
|Created|2014-09-26 22:31:25Z|
|Branch|https://github.com/twisted/twisted/tree/log-name-7652|

Per https://twistedmatrix.com/pipermail/twisted-python/2014-September/028799.html

<details><summary>Searchable metadata</summary>

```
trac-id__7652 7652
type__task task
reporter__glyph glyph
priority__normal normal
milestone__New_Logging_System New Logging System
branch__branches_log_name_7652 branches/log-name-7652
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__logger logger
keywords__ 
time__1411770685762470 1411770685762470
changetime__1481165825555494 1481165825555494
version__None None
owner__wsanchez wsanchez

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [43172]) Branching to logger-name-7652.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

(In [43261]) Branching to logger-name-7652-2.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph removed owner|
|-|-|

Buildbots look good.

As a bonus, since the buildbots are in considerably better state now than when this was originally landed, this takes the number of pyflakes and pydoctor errors in the package to zero, so the buildbots would not complain.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez commented|
|-|-|

Um, maybe wait on [#7596](https://github.com/twisted/twisted/issues/7596) to avoid merge hell?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

I doubt there will be much of a merge hell in either direction.  The file moves and whatnot should be tracked, and the changes are actually pretty minimal; when there are conflicts there should be a pretty trivial resolution.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

Let me put my money where my mouth is here - if there are merge conflicts, I volunteer to merge them, whichever of these lands first :-).

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)|@habnabit set owner to @habnabit|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

Changes looks good. Thanks!


Do we really need a link to NoneType.
Also, is there any Python developer confused by C{None} so that it needs a link to NoneType? I would say that C{None} is enough and that ``` @rtype: L{types.NoneType} ``` is just stating the obvious

```
         @return: C{None}
-        @rtype: L{NoneType}
+        @rtype: L{types.NoneType}
```

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_10"></a>|@glyph commented|
|-|-|

Replying to [adiroiban](#note_9):
> Changes looks good. Thanks!
> 
> 
> Do we really need a link to NoneType.

If we're going to talk about NoneType, then yes, it should be a `L{}`.

> Also, is there any Python developer confused by C{None} so that it needs a link to NoneType? I would say that C{None} is enough and that ``` @rtype: L{types.NoneType} ``` is just stating the obvious

In numerous preceding discussions we have arrived at a consensus that while `L{None}` is not particularly _necessary_, it is more _expressive_, as well as less ambiguous and more consistent with other parts of the documentation style.  So that's what we do.  (We might want to talk about getting rid of any `@type` annotation on a `@return: L{None}`  though…

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_11"></a>|@adiroiban commented|
|-|-|

I would like to talk about getting rid of @rtype for @return: L{None} :) ... so, is there any good reason to populate the docstring with this?

Does it add any valuable information to the docstring so that it is worth the effort to type it and to care about it?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_12"></a>|@glyph commented|
|-|-|

Replying to [adiroiban](#note_11):
> I would like to talk about getting rid of @rtype for @return: L{None} :) ... so, is there any good reason to populate the docstring with this?
> 
> Does it add any valuable information to the docstring so that it is worth the effort to type it and to care about it?

I'm inclined to agree that `@return: L{None}` should be sufficient.  I don't even think twistedchecker needs to be modified to honor this decision; am I remembering correctly that it's OK with a `@return` and no `@rtype`?

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_13"></a>|@adiroiban commented|
|-|-|

Is there anything left stopping this branch from being merged?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_14"></a>|@glyph set owner to @wsanchez|
|-|-|

It looks like this particular branch was branched from trunk.  This will need to be re-done on the logging branch itself, which wsanchez has been working on.  reassigning just to get his attention, but if someone else (for example habnabit) wants to take this over I'm sure he won't object :).

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_15"></a>|@wsanchez set status to `assigned`|
|-|-|

Yeah, I can deal with it.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_16"></a>|@wsanchez commented|
|-|-|

* BEAVER BAIT: Trash logs and debris. A log drive term.
* HIGHBALLER: A logger in a hurry. A fast, energetic worker.
* LANDING: The place to which logs are dragged for loading.
* LOKIE: The old-time steam locomotive, usually a Mallet or a Shay, that could run anyplace a logger could lay track. A logger could lay track anywhere.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_18"></a>|@wsanchez removed owner<br>@wsanchez set status to `new`|
|-|-|

to review.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_19"></a>|@ralphm set owner to @wsanchez|
|-|-|

Let's do it!

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_20"></a>|@wsanchez set status to `closed`|
|-|-|

(In [44439]) Merge branches/log-name-7652: rename twisted.python.logger to twisted.logger

Author: wsanchez
Reviewer: ralphm
Fixes: [#7652](https://github.com/twisted/twisted/issues/7652)
